sklearn/ensemble/ensemble_selection.py

```diff
@@ -0,0 +1,261 @@
+"""
+Ensemble Selection Classifier
+
+This module contains a Ensemble Selection Classifier for
+fitted classification estimators.
+
+"""
+
+from collections import Counter
+
+import operator
+import numpy as np
+
+from ..base import BaseEstimator
+from ..base import ClassifierMixin
+from ..utils import check_random_state
+from ..metrics import accuracy_score, f1_score
+from sklearn.utils.validation import check_is_fitted
+
+
+def _accuracy(y, proba):
+    return accuracy_score(y, np.argmax(proba, axis=1))
+
+def _f1(y, proba):
+    return f1_score(y, np.argmax(proba, axis=1))
+
+
+class EnsembleSelectionClassifier(BaseEstimator, ClassifierMixin):
+    """An ensemble classifier built by greedy stepwise selection.
+
+    Parameters
+    ----------
+    estimators : list of (string, estimator) tuples
+        The estimators from which the ensemble selection classifier is built.
+        These estimators must be fitted.
+
+    max_bag_estimators : integer, optional (default=50)
+        The maximum number of estimators at each bag ensemble selection
+        is terminated.
+
+    bag_fraction : float, optional (default=0.5)
+        Fraction of (post-pruning) models to randomly select for each bag
+        ensemble selection.
+
+    n_bags : integer, optional (default=20)
+        Number of bag ensemble selection.
+
+    n_best : integer, optional (default=1)
+        Number of the best estimators to be added into an ensemble
+        before each bag ensemble selection.
+
+    prune_fraction : float, optional (default=0)
+        Fraction of worst models to prune before each bag ensemble selection.
+
+    score_metric : {'accuracy', 'f1'}, optional (default='accuracy')
```

Comment:
Is there a reason not to use a `scoring` parameter with identical semantics to cross-validation utilities?